### PR TITLE
Remove Remaining Dependencies on jQuery UI

### PR DESCRIPTION
**Pull request by KTS915**
## Description
NOTE: PRs #1265, #1273, #1280, #1284, #1286, #1287, #1300, and #1307 should be merged before this one.

This PR removes core's remaining jQuery UI dependencies after those PRs have been applied. These are dependencies on functions within `jquery-ui-core`. In almost all cases, these functions have been replaced by vanilla JavaScript. In the case of `jquery-ui-widget`, however, it has been abstracted back into its own file (where it originally was before jQuery 1.12.1) so that it becomes a standalone dependency for `iris.js` and nothing more.

Once this PR has been successfully merged, we will have completely removed core's dependency on jQuery UI. We will then be able to deprecate usage of jQuery UI so that we could, in principle, remove jQuery UI (with the exception of the `jquery-ui-widget` file, which we'll then be treating effectively as a standalone library) from the ClassicPress bundle in CP v.3 (though whether we actually do that should be the subject of further discussion when v.3 is in preparation).

## Motivation and context
This completes the project to remove core's dependency on jQuery UI in order to replace it with vanilla JavaScript or native HTML. This makes for leaner code and faster, more efficient performance.

## How has this been tested?
On my localhost testing site on both Firefox and Chrome.

## Screenshots
Nothing to see: the changes are all "under the hood".

## Types of changes
New feature: no breaking changes as backward compatibility is maintained.

### CODE REVIEW

**Comment by KTS915** [comment]
The latest set of commits removes the last remaining dependency of core on `thickbox.js`. It is replaced with a native HTML5 dialog modal.

Use of the thickbox script can, therefore, be deprecated once this PR is merged. This is not a breaking change because the script remains in the CP bundle in case themes or plugins continue to use it, even though CP core no longer will.

The file `~/wp-admin/js/plugin-install.js` has now also been completely re-written in vanilla JavaScript, so it now has no dependencies at all, not even jQuery.

**Comment by KTS915** [comment]
It's perhaps worth noting that WP first considered removing thickbox (which this PR does) as long as 14 years ago! See https://core.trac.wordpress.org/ticket/10955 Some years later, another Trac ticket was also opened with the same objective: https://core.trac.wordpress.org/ticket/46985

On this PR, the following reported errors are all how that file was built originally, so re-writing them here is beyond the scope of this PR:
```
/home/runner/work/ClassicPress/ClassicPress/src/wp-admin/js/customize-controls.js
Error:   4010:5   error  Expected a function declaration  func-style
Error:   8859:9   error  Expected a function declaration  func-style
Error:   9039:11  error  Expected a function declaration  func-style
Error:   9161:8   error  Expected a function declaration  func-style
Error:   9253:9   error  Expected a function declaration  func-style
```

File: src/wp-admin/js/customize-controls.js [hunk 1]
```diff
@@ -3975,7 +4007,6 @@
 				statuses     = this.container.find('.dropdown-status'),
 				params       = this.params,
 				toggleFreeze = false,
-				// eslint-disable-next-line func-style
```
Comment (mattyrob) [inline src/wp-admin/js/customize-controls.js:3978]: `// eslint-disable-next-line func-style`

Leave this comment line in the code to silence `Expected a function declaration` errors / warnings in ESLint.

File: src/wp-admin/js/customize-controls.js [hunk 2]
```diff
@@ -8811,7 +8856,6 @@
 		// Bind site title display to the corresponding field.
 		if ( title.length ) {
 			api( 'blogname', function( setting ) {
-				// eslint-disable-next-line func-style
```
Comment (mattyrob) [inline src/wp-admin/js/customize-controls.js:8814]: `// eslint-disable-next-line func-style`

Leave this comment line in the code to silence `Expected a function declaration` errors / warnings in ESLint.

File: src/wp-admin/js/customize-controls.js [hunk 3]
```diff
@@ -8992,7 +9036,6 @@
 			api( settingId, function( setting ) {
 				$.each( o.controls, function( i, controlId ) {
 					api.control( controlId, function( control ) {
-						// eslint-disable-next-line func-style
```
Comment (mattyrob) [inline src/wp-admin/js/customize-controls.js:8995]: `// eslint-disable-next-line func-style`

Leave this comment line in the code to silence `Expected a function declaration` errors / warnings in ESLint.

File: src/wp-admin/js/customize-controls.js [hunk 4]
```diff
@@ -9115,7 +9158,6 @@
 
 		// Add behaviors to the static front page controls.
 		api( 'show_on_front', 'page_on_front', 'page_for_posts', function( showOnFront, pageOnFront, pageForPosts ) {
-			// eslint-disable-next-line func-style
```
Comment (mattyrob) [inline src/wp-admin/js/customize-controls.js:9118]: `// eslint-disable-next-line func-style`

Leave this comment line in the code to silence `Expected a function declaration` errors / warnings in ESLint.

File: src/wp-admin/js/customize-controls.js [hunk 5]
```diff
@@ -9208,7 +9250,6 @@
 		// Toggle visibility of Header Video notice when active state change.
 		api.control( 'header_video', function( headerVideoControl ) {
 			headerVideoControl.deferred.embedded.done( function() {
-				// eslint-disable-next-line func-style
```
Comment (mattyrob) [inline src/wp-admin/js/customize-controls.js:9211]: `// eslint-disable-next-line func-style`

Leave this comment line in the code to silence `Expected a function declaration` errors / warnings in ESLint.

File: src/wp-admin/js/theme-plugin-editor.js [hunk 6]
```diff
@@ -90,7 +90,6 @@ wp.themePluginEditor = (function( $ ) {
 	 * @return {void}
 	 */
 	component.showWarning = function() {
-		var oneSecond = 1000;
```
Comment (mattyrob) [inline src/wp-admin/js/theme-plugin-editor.js:93]: `var oneSecond = 1000;`

Leave this line in the code to silence `No magic number: 1000` errors / warnings in ESLint, with line 114 left as:

`}, oneSecond );`

File: src/wp-includes/script-loader.php [hunk 7]
```diff
@@ -1233,8 +1236,16 @@ function wp_default_scripts( $scripts ) {
...
 
-		$scripts->add( 'plugin-install', "/wp-admin/js/plugin-install$suffix.js", array( 'jquery', 'jquery-ui-core', 'thickbox' ), false, 1 );
+		$scripts->add( 'plugin-install', "/wp-admin/js/plugin-install$suffix.js", array(), false, 1 );
 		$scripts->set_translations( 'plugin-install' );
+		did_action( 'init' ) && $scripts->localize(
+			'plugin-install',
+			'pluginL10n',
+			array(
+				'close'            => __( 'Close' ),
+				'noiframes'        => __( 'This feature requires inline frames. You have iframes disabled or your browser does not support them.' )
```
Comment (mattyrob) [inline src/wp-includes/script-loader.php:1246]: To fix PHP error, there just needs to be a comma at the end of this line.

File: src/wp-includes/script-loader.php [hunk 8]
```diff
@@ -1254,7 +1265,10 @@ function wp_default_scripts( $scripts ) {
 
 		$scripts->add( 'farbtastic', '/wp-admin/js/farbtastic.js', array( 'jquery' ), '1.2' );
 
-		$scripts->add( 'iris', '/wp-admin/js/iris.min.js', array( 'jquery-ui-draggable', 'jquery-ui-slider', 'jquery-touch-punch' ), '1.1.1', 1 );
+		$scripts->add( 'iris', '/wp-admin/js/iris.min.js', array( 'jquery-ui-widget' ), '1.1.1', 1 );
```
Comment (mattyrob) [inline src/wp-includes/script-loader.php:1268]: Is changing the color picker not covered in another PR:
https://github.com/ClassicPress/ClassicPress/pull/1300

File: src/wp-admin/js/customize-controls.js (hunk 1 above)
Comment (KTS915) [reply src/wp-admin/js/customize-controls.js]: Thanks for the explanation! Now done.

File: src/wp-admin/js/customize-controls.js (hunk 2 above)
Comment (KTS915) [reply src/wp-admin/js/customize-controls.js]: Thanks for the explanation! Now done.

File: src/wp-admin/js/customize-controls.js (hunk 3 above)
Comment (KTS915) [reply src/wp-admin/js/customize-controls.js]: Thanks for the explanation! Now done.

File: src/wp-admin/js/customize-controls.js (hunk 4 above)
Comment (KTS915) [reply src/wp-admin/js/customize-controls.js]: Thanks for the explanation! Now done.

File: src/wp-admin/js/customize-controls.js (hunk 5 above)
Comment (KTS915) [reply src/wp-admin/js/customize-controls.js]: Thanks for the explanation! Now done.

File: src/wp-admin/js/theme-plugin-editor.js (hunk 6 above)
Comment (KTS915) [reply src/wp-admin/js/theme-plugin-editor.js]: Thanks for the explanation! Now done.

File: src/wp-includes/script-loader.php (hunk 8 above)
Comment (KTS915) [reply src/wp-includes/script-loader.php:1268]: Yes and no. In that PR, the dependency is changed to `jquery-ui-core`, which is correct after the other changes made in that PR. In this PR, though, I have abstracted out the one section of that file's code and put it in the code for the `jquery-ui-widget`, which used to be where it was in the first place. (Core currently still lists the widget file but redirects to the `jquery-ui-core` file.)

The alternative would be to leave the `jquery-ui-widget` file with just the widget code in it and continue to have the `jquery-ui-widget` file redirect to it, but I thought that that seemed unnecessarily messy.

File: src/wp-includes/script-loader.php (hunk 7 above)
Comment (KTS915) [reply src/wp-includes/script-loader.php]: Thanks for the explanation! Now done!

**Review by mattyrob — APPROVED** [review]

**Comment by mattyrob** [comment]
@KTS915 - We have a conflict in this PR after merging some of the other JS related PRs.

I think the correct code with this PR should be:
```
$scripts->add( 'iris', '/wp-admin/js/iris.min.js', array( 'jquery-ui-widget' ), '1.1.1', 1 );
```

Is that correct?

**Comment by KTS915** [comment]
@mattyrob Yes, that's correct!

**Pull request merged**